test/shared/src/main/scala/zio/test/Assertion.scala

```diff
@@ -543,6 +543,26 @@ object Assertion {
       }
     }
 
+  /**
+   * Makes a new assertion that verifies a TestFailure matches a given failure
+   */
+  final def testFails[E](failure: E): Assertion[TestFailure[E]] = {
```

Comment:
maybe `failsWith`? @adamgfraser wdyt?

Reply:
I think we can simplify this. I am reviewing now.